scriptworker/test/test_integration.py

```diff
@@ -172,30 +171,71 @@ def get_temp_creds(context):
 
 
 # run_maxtimeout {{{1
-async def maxtimeout_helper(context_function, partial_config, task_id):
+@pytest.mark.skipif(os.environ.get("NO_TESTS_OVER_WIRE"), reason=SKIP_REASON)
+@pytest.mark.parametrize("context_function", [get_context, get_temp_creds_context])
+@pytest.mark.asyncio
+async def test_run_maxtimeout(context_function):
+    task_id = slugid.nice().decode('utf-8')
+    partial_config = {
+        'task_max_timeout': 2,
+        'task_script': ('bash', '-c', '>&2 echo bar && echo foo && sleep 30 && exit 1'),
+    }
     async with context_function(partial_config) as context:
         result = await create_task(context, task_id, task_id)
         assert result['status']['state'] == 'pending'
         async with remember_cwd():
             os.chdir(os.path.dirname(context.config['work_dir']))
-            await worker.run_tasks(context, creds_key="integration_credentials")
+            status = await worker.run_tasks(context, creds_key="integration_credentials")
+            assert status == context.config['task_max_timeout_status']
+
+
+# cancel task {{{1
+async def do_cancel(context, task_id):
+    count = 0
+    while True:
+        await asyncio.sleep(1)
+        count += 1
+        assert count < 30, "do_cancel Timeout!"
+        if not context.task or not context.proc:
+            continue
+        await context.queue.cancelTask(task_id)
+        break
+
+
+async def run_task_to_cancel(context):
+    async with remember_cwd():
+        os.chdir(os.path.dirname(context.config['work_dir']))
+        status = await worker.run_tasks(context, creds_key="integration_credentials")
+        return status
 
 
 @pytest.mark.skipif(os.environ.get("NO_TESTS_OVER_WIRE"), reason=SKIP_REASON)
-@pytest.mark.parametrize("context_function", [get_context, get_temp_creds_context])
-def test_run_maxtimeout(event_loop, context_function):
+@pytest.mark.asyncio
+async def test_cancel_task():
     task_id = slugid.nice().decode('utf-8')
     partial_config = {
-        'task_max_timeout': 2,
+        'invalid_reclaim_status': 19,
+        'task_script': ('bash', '-c', '>&2 echo bar && echo foo && sleep 30 && exit 1'),
     }
-    pre = arrow.utcnow()
-    try:
-        event_loop.run_until_complete(maxtimeout_helper(context_function, partial_config, task_id))
-    except RuntimeError:
-        pass
-    post = arrow.utcnow()
-    # This may be flaky because claimWork can take up to 20 seconds..?
-    assert post.timestamp - pre.timestamp <= 8
+    # No temp_creds context because we won't have cancel scopes
```

Comment:
Could you expand on this comment? I'm not really sure what it means.

Reply:
*Why* don't they have cancel scopes?

Reply:
That's a question for the tc team. The temp creds are returned during the first `claimTask` call, and contain just enough scopes to run the task. That doesn't include cancel scopes. I believe that question is outside the scope of this PR.

Reply:
`claimTask` scopes docs at https://docs.taskcluster.net/docs/reference/platform/taskcluster-queue/references/api#claimTask

Reply:
I suppose we could add the cancel task scope in the `task.scopes`... not sure this comment is that noteworthy, but if you have suggestions, I can fix.

Reply:
I think maybe something like
> Don't use task-specific temporary credentials, since they don't allow us to cancel the created task.

or
> Don't use temporary credentials from `claimTask`, since they don't allow us to cancel the created task.

is clearer, since it gives some indication of where the temporary credentials are coming from, that we don't have control over them.  I was confused because I was assuming that we were somehow creating temporary credentials in the test.